Octokit/Clients/RepositoryCommitsClient.cs

```diff
@@ -105,6 +107,74 @@ public Task<CompareResult> Compare(long repositoryId, string @base, string head)
             return ApiConnection.Get<CompareResult>(ApiUrls.RepoCompare(repositoryId, @base, head));
         }
 
+        /// <summary>
+        /// Compare two references in a repository
+        /// </summary>
+        /// <param name="owner">The owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="base">The reference to use as the base commit</param>
+        /// <param name="head">The reference to use as the head commit</param>
+        /// <param name="options">Options for changing the API response</param>
+        [ManualRoute("GET", "/repos/{owner}/{repo}/compare/{base}...{head}")]
+        public Task<CompareResult> Compare(string owner, string name, string @base, string head, ApiOptions options)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(owner, nameof(owner));
+            Ensure.ArgumentNotNullOrEmptyString(name, nameof(name));
+            Ensure.ArgumentNotNullOrEmptyString(@base, nameof(@base));
+            Ensure.ArgumentNotNullOrEmptyString(head, nameof(head));
+            Ensure.ArgumentNotNull(options, nameof(options));
+
+            return Compare(ApiUrls.RepoCompare(owner, name, @base, head), options);
+        }
+
+        /// <summary>
+        /// Compare two references in a repository
+        /// </summary>
+        /// <param name="repositoryId">The Id of the repository</param>
+        /// <param name="base">The reference to use as the base commit</param>
+        /// <param name="head">The reference to use as the head commit</param>
+        /// <param name="options">Options for changing the API response</param>
+        [ManualRoute("GET", "/repositories/{id}/compare/{base}...{head}")]
+        public Task<CompareResult> Compare(long repositoryId, string @base, string head, ApiOptions options)
+        {
+            Ensure.ArgumentNotNullOrEmptyString(@base, nameof(@base));
+            Ensure.ArgumentNotNullOrEmptyString(head, nameof(head));
+            Ensure.ArgumentNotNull(options, nameof(options));
+
+            return Compare(ApiUrls.RepoCompare(repositoryId, @base, head), options);
+        }
+
+        private async Task<CompareResult> Compare(Uri uri, ApiOptions options)
+        {
+            var results = await ApiConnection.GetAll<CompareResult>(uri, options);
```

Comment:
Do you mind explaining the reason for this custom logic where it doesn't seem to exist for other comparisons? How is this tested?

Reply:
this method returns only one object (not a collection) but it contains properties with collections: `Commits` and `Files`. 
Without `ApiOptions` it returns up to 250 elements in this collections, but the API supports the pagination. I used an existing method `ApiConnection.GetAll` to work with pagination but a result of this method is a collection of the elements.
I've added a simple loop to aggregate data from collection to one object.

 in results we have a data structure like that 

```

[
	{
		<fields with the same values for all objects in list>,
		Commits:[0..100]
		Files: [0..100]
	},
	{
		<fields with the same values for all objects in list>,
		Commits:[101..200]
		Files: [101..200]
	},
	...
]

```

and we need to return an object like that
```
{
	<fields with the same values for all objects in list>,
	Commits:[0..end],
	Files: [0..end]
}
```

Reply:
I've added an integration test to check it: https://github.com/octokit/octokit.net/pull/2690/files#diff-0390f8750f37f9bd7484e089ecbab5421588cfd100243b64c6cab2bd0c6b20dcR302

Reply:
Understood, thanks!